filecoin_test.go

```diff
@@ -67,3 +75,90 @@ func TestFilecoinBasic(t *testing.T) {
 
 	qt.Assert(t, enc, qt.DeepEquals, fenc)
 }
+
+func generateFilecoinStateTreeStore() (cid.Cid, linking.LinkSystem) {
+	// The fixtures below were generated by running 40 iterations of
+	// https://github.com/filecoin-project/specs-actors/blob/d8d9867f68a3c299295efdc6d1b3421c9b63df57/actors/states/tree_test.go#L79-L94
+	_, filename, _, _ := runtime.Caller(0)
+
+	files, err := os.ReadDir(path.Join(path.Dir(filename), "fixtures"))
+	if err != nil {
+		panic(err)
+	}
+
+	store := cidlink.Memory{
+		Bag: make(map[string][]byte),
+	}
+
+	// preload the fixtures in the bag
+	for _, file := range files {
+		k, err := cid.Decode(file.Name())
+		if err != nil {
+			panic(err)
+		}
+
+		f, err := os.Open(path.Join(path.Dir(filename), "fixtures", file.Name()))
+		if err != nil {
+			panic(err)
+		}
+		data, err := io.ReadAll(f)
+		if err != nil {
+			panic(err)
+		}
+
+		store.Bag[string(k.Hash())] = data
+	}
+
+	ls := cidlink.DefaultLinkSystem()
+	ls.StorageReadOpener = store.OpenRead
+	ls.StorageWriteOpener = store.OpenWrite
+
+	// The Map root
+	root, err := cid.Decode("bafy2bzaceco2ynykenmzjht6mtmu2enaw5nfaxa5jhtpqjclecb7aqdygzqok")
+	if err != nil {
+		panic(err)
+	}
+	return root, ls
+}
+
+func TestFilecoinAdtMap(t *testing.T) {
+	root, ls := generateFilecoinStateTreeStore()
+	ndr, err := ls.Load(ipld.LinkContext{}, cidlink.Link{Cid: root}, hamt.HashMapNodePrototype.Representation())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	builder := hamt.FilecoinV3Prototype{}.NewBuilder().(*hamt.Builder)
+	builder = builder.WithLinking(ls, cidlink.LinkPrototype{})
+
+	nd := builder.Build().(*hamt.Node)
+
+	hmn := bindnode.Unwrap(ndr).(*hamt.HashMapNode)
+	if hmn == nil {
+		t.Fatal("invalid hashmap node")
+	}
+	nd.Hamt = *hmn
+
+	// id address #39
+	addrKey, err := hex.DecodeString("0027")
```

Comment:
This is such an excellent fixture, and I don't want to suggest that this isn't good enough, but as long as you have some interesting data perhaps you could extend this slightly, running this same block of logic a few more times:

* If you query `0001` you could get CallSeqNum of `1` which is in the middle of the other leaf
* If you queried `000c` then you could get CallSeqNum of `12` which is at the end of the root block (query of root block is kind of novel)